src/test/java/org/opensearch/ad/rest/AnomalyDetectorRestApiIT.java

```diff
@@ -304,9 +310,13 @@ public void testUpdateAnomalyDetector() throws Exception {
     }
 
     public void testUpdateAnomalyDetectorNameToExisting() throws Exception {
-        AnomalyDetector detector1 = createRandomAnomalyDetector(true, true, client());
+        AnomalyDetector detector1 = TestHelpers.randomAnomalyDetector(TestHelpers.randomUiMetadata(), null);
+        String indexName1 = detector1.getIndices().get(0);
+        TestHelpers.createIndex(client(), indexName1, TestHelpers.toHttpEntity("{\"name\": \"test\"}"));
 
-        AnomalyDetector detector2 = createRandomAnomalyDetector(true, true, client());
+        AnomalyDetector detector2 = TestHelpers.randomAnomalyDetector(TestHelpers.randomUiMetadata(), null);
+        String indexName2 = detector2.getIndices().get(0);
+        TestHelpers.createIndex(client(), indexName2, TestHelpers.toHttpEntity("{\"name\": \"test\"}"));
```

Comment:
Seems a lot of duplicate code in this test suite - suggest to create a helper method that creates a detector and ingests sample data to the generated source index